drivers/overlay/overlay.go

```diff
@@ -2554,18 +2568,11 @@ func (d *Driver) getAdditionalLayerPath(tocDigest digest.Digest, ref string) (st
 			ref = refElem
 		}
 		target := path.Join(ls.path, ref, tocDigest.String())
-		// Check if all necessary files exist
-		for _, p := range []string{
-			filepath.Join(target, "diff"),
-			filepath.Join(target, "info"),
-			filepath.Join(target, "blob"),
-		} {
-			if err := fileutils.Exists(p); err != nil {
-				wrapped := fmt.Errorf("failed to stat additional layer %q: %w", p, err)
-				return "", fmt.Errorf("%v: %w", wrapped, graphdriver.ErrLayerUnknown)
-			}
+		err := checkIfAllNesseseryFilesExistsInLayerStore(target)
+		if err == nil {
+			return target, nil
 		}
-		return target, nil
```

Comment:
@giuseppe @ktock “Additional layer stores” support multiple stores, but apparently the code was only able to work with the first one, since the very start of the feature ( https://github.com/containers/storage/commit/64f018103e1dba3de80e93a820b4ca50d9926ef6 ).

What do you think about simplifying to only support one store? Admittedly that would not help simplify the codebase all that much, ALS is AFAICS _only_ a feature within this driver and ~this one loop — and instead, we would need to add even more code to reject multiple stores when parsing options.

Either way, we seem not to have tests sufficiently covering the feature.